packages/usa-header/src/styles/_usa-megamenu.scss

```diff
@@ -1,14 +1,27 @@
 @use "sass:math";
-
+@use "sass:meta";
 @use "uswds-core" as *;
 
-@mixin outer-megamenu {
+@mixin outer-megamenu($header-max-width: $theme-header-max-width, $type) {
   @include u-pin("y");
   background-color: color("primary-darker");
   content: "";
   display: block;
   position: absolute;
-  width: 100%;
+
+  // retrieve the unit value of $theme-header-max-width, depending on type of value added
+  $mw: smart-quote($header-max-width);
+  @if meta.type-of($header-max-width) == "string" {
+    $mw: units($header-max-width);
+  }
```

Comment:
Checking for type in case user adds a unit value instead of a USWDS value. This maintains current capabilities.